src/components/FormMaxInputToggleButton.tsx

```diff
@@ -0,0 +1,47 @@
+import styled from 'styled-components';
+
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { formMixins } from '@/styles/formMixins';
+
+import { Icon, IconName } from './Icon';
+import { ToggleButton } from './ToggleButton';
+
+type ElementProps = {
+  size?: ButtonSize;
+  isInputEmpty: boolean;
+  isLoading: boolean;
+  onPressedChange: (isPressed: boolean) => void;
+};
+
+export const FormMaxInputToggleButton = ({
+  size = ButtonSize.XSmall,
+  isInputEmpty,
+  isLoading,
+  onPressedChange,
+}: ElementProps) => {
+  const stringGetter = useStringGetter();
+
+  return (
+    <$FormMaxInputToggleButton
+      size={size}
+      isPressed={!isInputEmpty}
+      disabled={isLoading}
+      onPressedChange={onPressedChange}
+      shape={isInputEmpty ? ButtonShape.Rectangle : ButtonShape.Circle}
+    >
+      {isInputEmpty ? stringGetter({ key: STRING_KEYS.MAX }) : <Icon iconName={IconName.Close} />}
+    </$FormMaxInputToggleButton>
+  );
+};
+
+const $FormMaxInputToggleButton = styled(ToggleButton)`
+  ${formMixins.inputInnerToggleButton}
+
+  svg {
```

Comment:
minor nit: current the close button's 'x' looks pretty small -- can we bump it up / change font size or reduce padding?

Reply:
<img width="108" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/78913268-36ff-49b0-8ea1-c35bf70a9a60">
and perhaps add a border matching whatever 'max' button has?